DatadogCore/Tests/TestsObserver/DatadogTestsObserver.swift

```diff
@@ -132,34 +132,6 @@ internal class DatadogTestsObserver: NSObject, XCTestObservation {
 
             If all above conditions are met, this failure might indicate a memory leak in the implementation.
             """
-        ),
```

Comment:
Removing these checks seems like a back step - in case of a memory leak we will not able to catch any lingering swizzlings.

A check something like if there is an instance of swizzler in memory or not will be helpful.

something like 
```swift
func checkInstancesOfType(_ type: AnyClass) -> Bool {
    let typeString = String(describing: type)
    
    for bundle in Bundle.allBundles {
        if let className = bundle.classNamed(typeString) {
            // Check if an instance of the class exists
            if class_getInstanceSize(className) > 0 {
                return true
            }
        }
    }
    
    return false
}
```

Reply:
It is already there, see [my comment](https://github.com/DataDog/dd-sdk-ios/pull/1588#discussion_r1433930181). These checks were redundant, we are already asserting that no swizzling is left [here](https://github.com/DataDog/dd-sdk-ios/blob/738d599b46f982275a6c41c48578808403f36eb6/DatadogCore/Tests/TestsObserver/DatadogTestsObserver.swift#L35L44). Failure will print the method signature, so we will know which sizzling is at fault.